planetary-variables/soil-water-content/soil-water-content-backward-average/raw.js

```diff
@@ -0,0 +1,57 @@
+//VERSION=3
+
+const nDays = 20;           // The number of days to load data for
+const scaleFactor = 1000;   // The scale factor for the SWC values
+
+function setup() {
+    return {
+      input: ["SWC", "dataMask"],
+      output: { bands: 1, sampleType: "FLOAT32" },
+      mosaicking: "ORBIT"
+    };
+  }
+  
+  function preProcessScenes(collections) {
+    collections.scenes.orbits = collections.scenes.orbits.filter(function (orbit) {
+      var orbitDateFrom = new Date(orbit.dateFrom)
+      // Select all images within the last nDays
+      return orbitDateFrom.getTime() >= (collections.to.getTime() - (nDays * 24 * 3600 * 1000));
+    })
+    return collections
+  }
+  
+  function get_mean_swc_value(swc, dataMask) {
+    // Get the sum of all SWC values
+    let n_valid_dates = 0;
+    let sum = 0;
+    for (let i = 0; i < swc.length; i++) {
+        if (dataMask[i]) {
+            sum += swc[i];
+            n_valid_dates += 1;
+        }
+    }
+    
```

Comment:
To avoid those two `.map()` calls you can just access the appropriate values in the loop directly:

```suggestion
  function get_mean_swc_value(samples) {
    // Get the sum of all SWC values
    let n_valid_dates = 0;
    let sum = 0;
    for (let i = 0; i < samples.length; i++) {
        if (samples[i].dataMask) {
            sum += samples[i].swc;
            n_valid_dates += 1;
        }
    }
    
```